### PR TITLE
e2e: Upgrade Ginkgo to V2 version. 

### LINKED ISSUE #344: [feature] Improve E2E test framework.
**Is your feature request related to a problem? Please describe.**

- [x] Basic framework, including automatic creation, cleanup API.
- [x] MySQLCluster related API.
- [ ] Automatically prepare data/run test through Sysbench.
- [ ] Chaosmesh related API.
- [ ] CI.

Test cases:

- [ ] HA
- [ ] Upgrade
- [ ] Scale
- [ ] Chaos(net, kill, io...)
- [ ] Backup
- [ ] User

 
**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### LINKED ISSUE #358: [feature] Optimize test cases filtering methods and log collection methods.
**Is your feature request related to a problem? Please describe.** 

The Ginkgo V1 version cannot be categorized and organized well, for example, test Replicas (2-3-5-3-2), must be placed in an IT (), otherwise it will be executed in random order. This can lead to this type of CASE only execute at the same time, and cannot be tested separately.

**Describe the solution you'd like**

Upgrade to V2 versions, it can implement the case order execution, and support the use of tag classification.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by runkecheng**
### What type of PR is this?

/enhancement

### Which issue(s) this PR fixes?

Fixes #358

### What this PR does?

Summary:

- Upgrade Ginkgo to V2 version, replace the API.

- Each suite`s logs is stored in a separate folder and outputs the test overview.

- POD logs of different use cases are output to different files.

- Wait for the NS to remove and then exit after completing the test.

- Configure unified management.

- Add a label for each level of the use case.

- Example of adding screening cases in documentation.

### CODE REVIEW

**Comment by runkecheng** [comment]
https://github.com/onsi/ginkgo/issues/874

File: test/e2e/README.md [hunk 1]
```diff
@@ -14,48 +14,26 @@ export KUBECONFIG=$HOME/.kube/config
 
 ### Step 2: Run test
 
+- Running all cases
```
Comment (andyli029) [inline test/e2e/README.md:17]: Add ginkgo version

File: test/e2e/framework/test_context.go [hunk 2]
```diff
@@ -19,9 +19,44 @@ package framework
...
+const (
+	// The namespace where the resource created by E2E.
+	RadondbMysqlE2eNamespace = "radondb-mysql-e2e"
+	// The name of the Operator to create.
+	OperatorReleaseName = "e2e-test"
+	// Export POD logs and test overview.
+	DumpLogs = true
+	// Optional directory to store junit and pod logs output in.
+	// If not specified, no junit or logs files will be output.
+	ReportDir = "e2e_logs"
```
Comment (andyli029) [inline test/e2e/framework/test_context.go:39]: ReportDir
->
ReportDirPrefix

File: test/e2e/e2e_test.go [hunk 3]
```diff
@@ -17,18 +17,165 @@ limitations under the License.
...
+
+	if err := c.Create(context.TODO(), operatorNsObj); err != nil {
+		if !strings.Contains(err.Error(), "already exists") {
+			Fail(fmt.Sprintf("can't create mysql-operator namespace: %s", err))
+		}
+	}
+
+	if framework.TestContext.DumpLogs {
+		By("Create log dir")
+		os.MkdirAll(fmt.Sprintf("%s_%d", framework.TestContext.ReportDirPrefix, GinkgoRandomSeed()), 0777)
```
Comment (acekingke) [inline test/e2e/e2e_test.go:80]: dir maybe do not random， use date time maybe reasonable.

File: test/e2e/e2e_test.go [hunk 4]
```diff
@@ -17,18 +17,165 @@ limitations under the License.
...
+	if err := framework.DeleteNS(client, framework.RadondbMysqlE2eNamespace, framework.DefaultNamespaceDeletionTimeout); err != nil {
+		framework.Failf(fmt.Sprintf("Can't delete namespace: %s", err))
+	}
+}, func() {
+	framework.Logf("Running AfterSuite actions on node 1")
+})
+
+var _ = ReportAfterSuite("Collect log", func(report Report) {
+	if framework.TestContext.DumpLogs {
+		f, err := os.OpenFile(path.Join(fmt.Sprintf("%s_%d", framework.TestContext.ReportDirPrefix, GinkgoRandomSeed()), "overview.txt"), os.O_RDWR|os.O_CREATE, 0644)
```
Comment (acekingke) [inline test/e2e/e2e_test.go:118]: the same to above

File: test/e2e/README.md [hunk 5]
```diff
@@ -14,48 +14,28 @@ export KUBECONFIG=$HOME/.kube/config
 
 ### Step 2: Run test
 
+> The Ginkgo version of the following examples is V2.
```
Comment (andyli029) [inline test/e2e/README.md:17]: Add reference about ginkgo.

**Review by andyli029 — CHANGES_REQUESTED** [review]
Ping

**Review by andyli029 — APPROVED** [review]

**Pull request merged**